build_web_compilers/lib/src/dev_compiler_bootstrap.dart

```diff
@@ -338,12 +368,27 @@ require.config({
 });
 
 requirejs.onResourceLoad = function (context, map, depArray) {
-  // TODO Handle dynamic changes in module dependancies
+  if (\$dartLoader.modulesGraph.has(map.name)) {
+    // TODO Move this logic to better place
+    var previousDeps = \$dartLoader.modulesGraph.get(map.name);
+    var changed = previousDeps.length != depArray.length;
+    changed = changed || depArray.some(function(dep) {
+      return !previousDeps.includes(dep.name);
+    });
+    if (changed) {
+      console.warn("Dependencies graph change for module '" + map.name + "' detected. " + 
+        "Dependencies was [" + previousDeps + "], now [" +  depArray.map((dep) => dep.name) +"]. " +
+        "Page can't be hot-reloaded, firing full page reload.");
+      window.location.reload();
+    }
+  }
+  \$dartLoader.modulesGraph.set(map.name, []);
```

Comment:
should this all move into an else block? I don't think we need to clear it out and set it up again.